web/src/pages/onboarding.tsx

```diff
@@ -301,42 +313,27 @@ const OnboardingPage = (props: Props): ReactElement => {
     if (!updateQueue) return;
 
     setRegistrationDimension("Onboarded Guest");
-    if (isAdditionalBusiness) {
-      analytics.event.onboarding_last_step_save_additional_business_button.click.finish_additional_business_onboarding();
-    } else {
-      analytics.event.onboarding_last_step.submit.finish_onboarding();
-    }
 
-    const isRemoteSellerWorker = isRemoteWorkerOrSellerBusiness({
+    isAdditionalBusiness
+      ? analytics.event.onboarding_last_step_save_additional_business_button.click.finish_additional_business_onboarding()
+      : analytics.event.onboarding_last_step.submit.finish_onboarding();
+
+    updateQueue.queueBusiness({
       ...updateQueue.currentBusiness(),
       profileData: newProfileData,
+      onboardingFormProgress: "COMPLETED",
     });
 
-    let newUserData: UserData = {
-      ...updateQueue.current(),
-      businesses: {
-        ...updateQueue.current().businesses,
-        [updateQueue.current().currentBusinessId]: {
-          ...updateQueue.currentBusiness(),
-          profileData: {
-            ...newProfileData,
-            operatingPhase: isRemoteSellerWorker
-              ? OperatingPhaseId.GUEST_MODE_WITH_BUSINESS_STRUCTURE
-              : newProfileData.operatingPhase,
-          },
-          onboardingFormProgress: "COMPLETED",
-        },
-      },
-    };
-
-    newUserData = await api.postGetAnnualFilings(newUserData);
-    updateQueue.queue(newUserData);
+    if (isRemoteWorkerOrSellerBusiness(updateQueue.currentBusiness()) && OperatingPhaseId.GUEST_MODE) {
+      updateQueue.queueProfileData({ operatingPhase: OperatingPhaseId.GUEST_MODE_WITH_BUSINESS_STRUCTURE });
+    }
 
-    if (newProfileData.legalStructureId) {
-      const completed: TaskProgress = "COMPLETED";
-      updateQueue.queueTaskProgress({ [businessStructureTaskId]: completed });
+    if (updateQueue.currentBusiness().profileData.legalStructureId) {
+      updateQueue.queueTaskProgress({ [businessStructureTaskId]: "COMPLETED" });
     }
 
+    const newUserData = await api.postGetAnnualFilings(updateQueue.current());
```

Comment:
I moved this logic down since we are waiting for an api call. All of the above logic is set in the queue.